src/core/core.controller.js

```diff
@@ -627,8 +627,13 @@ module.exports = function(Chart) {
 			}
 
 			// On Hover hook
-			if (hoverOptions.onHover) {
-				hoverOptions.onHover.call(me, me.active);
+			if (e.type === 'mousemove') {
+			    if (hoverOptions.onHover) {
+			        hoverOptions.onHover.call(me, me.active);
+			    }
+			    if (me.legend && me.legend.handleEvent && me.legend.options.onHover) {
```

Comment:
Agree, the extra calculations should not be done if useless, but this is the responsibility of the legend, not the controller (since this logic is in defined by the legend). I updated my previous comment accordingly.